src/Network/GRPC/Client/Connection.hs

```diff
@@ -165,13 +166,18 @@ data ReconnectPolicy =
     -- connection), do not attempt to connect again.
     DontReconnect
 
-    -- | Reconnect after random delay after the IO action returns
+    -- | Reconnect to the (potentially different) server after the IO action
+    -- returns
+    --
+    -- If the 'Maybe' is 'Just', we'll attempt to reconnect to a server at the
+    -- new address. If 'Nothing', we'll attempt to connect to the original
+    -- server that 'withConnection' was given.
```

Comment:
The documentation is entirely in line with the implementation, but on first reading I nonetheless misunderstood. I thought on `Nothing` we'd reconnect to the previous server we were connected to, but instead you choose to connect to the _initial_ server. I think both choices are defensible, not sure what the most sensible default is. Perhaps instead of a `Maybe` we should have 

```haskell
data ReconnectTo =
    ReconnectToPrev
  | ReconnectToOrig
  | ReconnectToNew Server
```

or something along these lines?